core/src/main/java/ch/unibas/dmi/dbis/polyphenydb/Store.java

```diff
@@ -47,4 +53,94 @@ public Store( final int storeId, final String uniqueName ) {
     public abstract void updateColumnType( Context context, CatalogColumn catalogColumn );
 
     public abstract String getAdapterName();
+
+    public abstract List<AdapterSetting> getAvailableSettings();
+
+    public abstract void shutdown();
+
+    protected abstract void applySetting( AdapterSetting s, String newValue );
+
+    public void updateSettings( Map<String, String> newSettings ) {
+        for ( AdapterSetting s : getAvailableSettings() ) {
+            if ( newSettings.containsKey( s.name ) ) {
+                if ( s.modifiable ) {
+                    String newValue = newSettings.get( s.name );
+                    if ( !s.canBeNull && newValue == null ) {
+                        throw new RuntimeException( "Setting \"" + s.name + "\" is not allowed to be null!" );
+                    }
+                    if ( newValue.equals( settings.get( s.name ) ) ) {
```

Comment:
Shouldn't this be negated?

```suggestion
                    if ( ! newValue.equals( settings.get( s.name ) ) ) {
```

Reply:
Oh yes... Thx